tservice/src/seal.rs

```diff
@@ -20,6 +18,39 @@ const DEFAULT_MISCELLANEOUS_MASK_FOR_SEAL: u32 = 0xF0000000;
 const DEFAULT_ATTRIBUTES_FLAGS_FOR_SEAL: u64 = 0xFF0000000000000B;
 const DEFAULT_KEY_POLICY_FOR_SEAL: KeyPolicy = KeyPolicy::MRSIGNER;
 
+pub type Result<T> = CoreResult<T, Error>;
+
+#[derive(Clone, Debug, displaydoc::Display, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[non_exhaustive]
+pub enum Error {
+    /// Error from SGX function {0}
+    Sgx(mc_sgx_core_types::Error),
+    /// FFI error {0}
+    Ffi(mc_sgx_core_types::FfiError),
+    /// Failed calculating the sealed data size, `sgx_calc_sealed_data_size()`
+    CalculateSealedDataSize,
+    /** Failed getting the decrypted data size `sgx_get_encrypt_txt_len()`
+     *  (Yes it is `decrypted` data size with a function named `encrypt`) */
```

Comment:
Up for ideas on this, it feels like a misnomer in the SGX SDK, but I think it's nice to call out the function name.

Reply:
error removed now